{
  Throwable cause;
  try {
    T result=get();
    callbacks.success(result);
    return;
  }
 catch (  InterruptedException ex) {
    Thread.currentThread().interrupt();
    return;
  }
catch (  ExecutionException ex) {
    cause=ex.getCause();
    if (cause == null) {
      cause=ex;
    }
  }
catch (  Throwable t) {
    cause=t;
  }
  callbacks.failure(cause);
}
